app/src/main/java/com/cornellappdev/uplift/ui/screens/ClassDetailScreen.kt

```diff
@@ -0,0 +1,336 @@
+package com.cornellappdev.uplift.ui.screens
+
+import android.content.Intent
+import android.provider.CalendarContract
+import android.util.Log
+import androidx.compose.foundation.*
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Icon
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.core.content.ContextCompat.startActivity
+import androidx.lifecycle.viewmodel.compose.viewModel
+import coil.compose.AsyncImage
+import com.cornellappdev.uplift.R
+import com.cornellappdev.uplift.models.UpliftClass
+import com.cornellappdev.uplift.ui.components.ClassInfoCard
+import com.cornellappdev.uplift.ui.viewmodels.ClassDetailViewModel
+import com.cornellappdev.uplift.util.PRIMARY_BLACK
+import com.cornellappdev.uplift.util.bebasNeueFamily
+import com.cornellappdev.uplift.util.montserratFamily
+
+
+/**
+ * A screen that displays details for a given [UpliftClass].
+ */
+@Composable
+fun ClassDetailScreen(
+    classDetailViewModel: ClassDetailViewModel = viewModel()
+) {
+    val upliftClass by classDetailViewModel.classFlow.collectAsState()
+    val context = LocalContext.current
+
+    val scrollState = rememberScrollState()
+
+    Log.d("url", upliftClass?.imageUrl.toString())
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .verticalScroll(scrollState)
+    ) {
+        // Top Part
+        Box(modifier = Modifier
+            .fillMaxWidth()
+            .graphicsLayer {
+                translationY = 0.5f * scrollState.value
+            }
+        ) {
+            AsyncImage(
+                model = upliftClass?.imageUrl,
+                contentDescription = null,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .aspectRatio(1f)
+                    .graphicsLayer {
```

Comment:
is this the modifier that allows the white out transitions of the image?

Reply:
Yes, `graphicsLayer` allows for the parallax scrolling... it can do a good amount more that I'm not too familiar with too--I've only used it a handful of times ever.